src/main/java/uk/gov/justice/digital/delius/service/OffenderManagerService.java

```diff
@@ -144,8 +144,10 @@ private CommunityOrPrisonOffenderManager allocatePrisonOffenderManager(final Pro
                         newPrisonOffenderManager.addResponsibleOfficer(responsibleOfficerRepository.save(responsibleOfficerOf(offender, newPrisonOffenderManager)));
                         contactService.addContactForResponsibleOfficerChange(newPrisonOffenderManager, existingPOM);
                     });
+            offender.getPrisonOffenderManagers().remove(existingPOM);
         }, () -> contactService.addContactForPOMAllocation(newPrisonOffenderManager));
 
+        offender.getPrisonOffenderManagers().add(newPrisonOffenderManager);
```

Comment:
ensure Offender is in the correct state for when JPA reuses it in a later transaction